sympy/matrices/random.py

```diff
@@ -0,0 +1,2378 @@
+
+# === fundamental constructor ===
+
+
+def super_elementary_matrix(dim,
+                            index=None,
+                            value=None,
+                            *scalars):
+    r"""super elementary matrix n x n, i.e. identiy with a 2 x 2 block
+
+    Explanation
+    ===========
+    The super elementary matrix $A$ is a generalization of elementary matrices
+    as well as of a rotation matrices that rotate only a single plane.
+
+    In two dimensions any invertible matrix
+
+    .. math::
+
+       A = \left[\begin{array}{cc}a & b \\ -c & d\end{array}\right]
+
+    is super elementary, i.e. $ad+bc \neq 0$. In higher dimensions,
+    any matrix $A$ looking like itdentiy matrix but with entries
+
+    .. math::
+
+        A[i,i] = a, \quad A[i,j] = b, \quad A[j,i] = -c, \quad A[j,j] = d
+
+    and $ad + bc \neq 0$.
+
+    This includes elementary matrices of matrix operation
+    for Gauss elimination. So multiplication with $A$ gives for
+
+    * $a=0=d$ and $b=1=-c$ a *transposition*, i.e. swapping rows $i$ and $j$
+    * $a=1$, $d=\lambda$ and $b=0=-c$ this matrix describes scaling the
+      row $j$ by $\lambda$
+    * $a=1=d$, $b=\lambda$, $-c=0$ adding the $\lambda$ multiple
+      of the row $i$ to row $j$.
+
+    Moreover, a simple rotation by $\phi$ in $(i,j)$ plane is super elementary,
+    given by $a=\cos \phi = d$ and $b = \sin \phi = c$. Hence,
+
+    .. math::
+
+       \left[\begin{array}{cc}a & b \\ -c & d\end{array}\right]
+       =
+       \left[\begin{array}{cc}
+        \cos \phi & \sin \phi \\ -\sin \phi & \cos \phi \end{array}\right]
+
+    In this module the super elementary matrix serves as the base class to
+    create futher matrices of given type.
+
+    Examples
+    ========
+
+    >>> from sympy.matrices.random import super_elementary_matrix
+
+    >>> super_elementary_matrix(3, (0,1))
+    Matrix([
+    [0, 1, 0],
+    [1, 0, 0],
+    [0, 0, 1]])
+
+    >>> super_elementary_matrix(3, (0,2), value=5)
+    Matrix([
+    [1, 0, 5],
+    [0, 1, 0],
+    [0, 0, 1]])
+
+    >>> super_elementary_matrix(3, (2,2), value=5)
+    Matrix([
+    [1, 0, 0],
+    [0, 1, 0],
+    [0, 0, 5]])
+
+    >>> super_elementary_matrix(3, (1,2), 1,2,3,4)
+    Matrix([
+    [1, 0, 0],
+    [0, 1, 2],
+    [0, -3, 4]])
+
+    Parameters
+    ==========
+    dim : integer
+        dimension of matrix
+    index : tuple(int, int)
+        coordinates (i,j) of super elementary square
+    value : symbol (optional)
+        value as elementary entry
+    *scalars : symbols (optional)
+        up to three additional values as additional super elementry entries
+
+    See Also
+    ========
+    diagonal
+    elementary
+    transposition
+    rotation
+
+    """
+    obj = _eye(dim)
+    if index is None:
+        return obj
+
+    if not isinstance(index, (list, tuple, set)) or not len(index) == 2:
+        raise ValueError(
+            "index argument must be tuple of two matrix index integer.")
+    row, col = index
```

Comment:
I don't think that a set should be allowed here if the intention is to iterate over index.

Reply:
removed `set` here.